File: 프로그래머스/1/77884. 약수의 개수와 덧셈/약수의 개수와 덧셈.c
#include <stdio.h>
#include <stdbool.h>
#include <stdlib.h>

int solution(int left, int right) {
    int answer = (right + left)*(right - left + 1)/2;
    for(int i=1;i<=31;i++){
        int t = i*i;
        if(t>=left && t<=right){
            answer -= t;
            answer -= t;
        }
    }
    return answer;
}

File: 프로그래머스/1/77884.약수의 개수와 덧셈-Python/약수의 개수와 덧셈.py
def solution(left, right):
    answer = 0
    l, r = 0, 1
    for i in range(32):
        if i **2 >= left:
            l = i
            break
    for i in range(32):
        if i**2 <= right:
            r = i

    answer = (right - left + 1)* (right+ left)//2
    for i in range(l,r+1):
        answer -= 2*i**2

    return answer